controllers/accountControllers/userAccountController.js

```diff
@@ -1,117 +1,110 @@
-const {Account} = require('../../models/userAccountModel');
-const {User} = require('../../models/userModel');
-const {validateAccount} = require('../../validation/validationAccount');
-
-
-const createAccount =  async (req, res, next) => {
-    const { error } = await validateAccount.validate(req.body);
-    if(error){
-        return next (new Error(error));
-    }
-    const {
-        accountName, 
-        accountCurrency, 
-        accountAmount, 
-        accountDescription, 
-        accountStat
-    }   = req.body;
-
-    const isAccount = await Account.findOne({accountName});
-
-    if(isAccount){
-        return next (new Error(`${isAccount.accountName} already exists.`));
-    }
-
-    const userID = req.user.id;
-    const account = await Account.create({
-        user: userID,
-        accountName, 
-        accountCurrency, 
-        accountAmount, 
-        accountDescription, 
-        accountStat,
-        accountCreated: new Date().toGMTString()
+const { Account } = require('../../models/userAccountModel');
+const { User } = require('../../models/userModel');
+const { validateAccount } = require('../../validation/validationAccount');
+
+const createAccount = async (req, res, next) => {
+  const { error } = await validateAccount.validate(req.body);
+  if (error) {
+    return next(new Error(error));
+  }
+  const {
+    accountName,
+    accountCurrency,
+    accountAmount,
+    accountDescription,
+    accountStat,
+  } = req.body;
+
+  const isAccount = await Account.findOne({ accountName });
+
+  if (isAccount) {
```

Comment:
Such validation can be done in mongoose schema

Reply:
I have questions regarding validation. 
Should I make validation only in mongoose, or is it better to do it with both Joi package and mongoose on BE? 
What about front-end validation? If I have front-end validation should I use three of them for safety or isn't it necessary?

Reply:
It depends on case, for example `required` can be validated only on mongoose. Maintaining several different validations is challenging. Frontent validation is required